Extensions/TileMap/simpletilemapruntimeobject.ts

```diff
@@ -0,0 +1,764 @@
+      for (const property of initialInstanceData.stringProperties) {
+        if (property.name === 'tilemap') {
+          this._initialTileMapAsJsObject = JSON.parse(property.value);
+        }
+      }
+
+      // 2. Update the renderer so that it updates the tilemap object
+      // (used for width and position calculations).
+      this._loadInitialTileMap((tileMap: TileMapHelper.EditableTileMap) => {
+        // 3. Set custom dimensions if applicable.
+        if (initialInstanceData.customSize) {
+          this.setWidth(initialInstanceData.width);
+          this.setHeight(initialInstanceData.height);
+        }
+
+        // 4. Update position (calculations based on renderer's dimensions).
+        this._renderer.updatePosition();
+
+        this._collisionTileMap = new gdjs.TileMap.TransformedCollisionTileMap(
+          tileMap,
+          this._hitBoxTag
+        );
+        this.updateTransformation();
+      });
+    }
+
+    private _loadInitialTileMap(
+      tileMapLoadingCallback: (tileMap: TileMapHelper.EditableTileMap) => void
+    ): void {
+      if (!this._initialTileMapAsJsObject) return;
+
+      this._tileMapManager.getOrLoadSimpleTileMap(
+        this._initialTileMapAsJsObject,
+        this.name,
+        this._tileSize,
+        this._columnCount,
+        this._rowCount,
+        (tileMap: TileMapHelper.EditableTileMap) => {
+          this._initialTilesWithHitBox.forEach((tileId) => {
+            const tileDefinition = tileMap.getTileDefinition(tileId);
+            if (!tileDefinition) {
+              console.warn(
+                `Could not set hit box for tile with id ${tileId}. Continuing.`
+              );
+              return;
+            }
+            tileDefinition.addHitBox(this._hitBoxTag, [
+              [0, 0],
+              [0, tileMap.getTileHeight()],
+              [tileMap.getTileWidth(), tileMap.getTileHeight()],
+              [tileMap.getTileWidth(), 0],
+            ]);
+          });
+
+          this._tileMapManager.getOrLoadSimpleTileMapTextureCache(
+            (textureName) => {
+              return (this.getInstanceContainer()
+                .getGame()
+                .getImageManager()
+                .getPIXITexture(textureName) as unknown) as PIXI.BaseTexture<
+                PIXI.Resource
+              >;
+            },
+            this._atlasImage,
+            this._tileSize,
+            this._columnCount,
+            this._rowCount,
+            (textureCache: TileMapHelper.TileTextureCache | null) => {
+              if (!textureCache) {
+                // getOrLoadTextureCache already log warns and errors.
+                return;
+              }
+              this._renderer.updatePixiTileMap(tileMap, textureCache);
+              tileMapLoadingCallback(tileMap);
+            }
+          );
+        }
+      );
+    }
+
+    onDestroyed(): void {
+      super.onDestroyed();
+      this._renderer.destroy();
+    }
+
+    setWidth(width: float): void {
```

Comment:
Is the code about transformation mostly the same as the other object?
Do you think an abstract class would be easier to maintain (maybe it would be worse I don't know)?

Reply:
ok